### PR TITLE
chore: artifact and package cleanup

### PR DESCRIPTION
**Pull request by diervo**
### Cleanup all files towards npm publishing.
We now only push the neccesary files on the pack command.

### Refactoring the way we generate artifacts
Packages: engine, synthetic-shadow and wire will only generate the module version.

### New LWC package with better bundle artifacts
A new package LWC is now responsible to generate and embed all four LWC core packages (compiler, engine, synthetic, wire). We know generate iife, which is superior that the UMD for many cases.

## Does this PR introduce breaking changes?
* 🚨 `Yes, it does introduce breaking changes.`

#### Breaking changes:
- We have changed the name of the engine when used standalone as a UMD or IIFE. So instead of a global called `Engine` now its called `LWC`

- We have changed the artifacts produced on the packages.

## Usage

```js
const { getModulePath, getModule } = require('lwc');
const LWC_ENGINE = getModulePath('engine', 'iife', 'es2017', 'dev');
const LWC_ENGINE_COMPAT = getModulePath('engine', 'iife', 'es5', 'dev');
```

### CODE REVIEW

**Comment by diervo** [comment]
This change is likely to break your build process.
@tedconn @reiniergs @muenzpraeger @nrkruk @dbleakley 

You should only need to import "lwc" rather than @lwc/engine...
See the details.

File: packages/@lwc/babel-plugin-component/package.json [hunk 1]
```diff
@@ -16,8 +21,5 @@
     },
     "devDependencies": {
         "common-tags": "^1.8.0"
-    },
-    "publishConfig": {
-        "access": "public"
```
[the remark this answers is not in the document]
Comment (diervo) [reply packages/@lwc/babel-plugin-component/package.json]: Good catch I though default was public but for scope apparently is restrictive

File: packages/@lwc/engine/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+// eslint-disable-next-line no-console
+console.log('@lwc/engine cannot be imported via cjs');
```
Comment (caridy) [inline packages/@lwc/engine/index.js]: hmm, why not?

**Comment by caridy** [comment]
I'm definitely not confortable with pkgs that will be pushed to npm and can't be imported using `require()` that seems bad.

My initial assumption was that we will have cjs (via ts lib folder) per pkg, and modules via (ts or rollup) per module. Then a pkg that does integration, and can be used easily with all the modes, and directly into a browser via a url, etc. That doesn't seem to be the case now.

**Comment by caridy** [comment]
LGTM, still test failures

**Pull request merged**